compiler-plugin/src/main/java/io/ballerina/stdlib/jwt/compiler/staticcodeanalyzer/JWTCipherAlgorithmAnalyzer.java

```diff
@@ -0,0 +1,158 @@
+                            getDocument(context),
+                            context.node().location(),
+                            JWTRule.AVOID_WEAK_CIPHER_ALGORITHMS.getId()
+                    );
+                }
+            } else if (expr instanceof SimpleNameReferenceNode varRef) {
+                if (isInsecureVariableReference(varRef)) {
+                    reporter.reportIssue(
+                            getDocument(context),
+                            context.node().location(),
+                            JWTRule.AVOID_WEAK_CIPHER_ALGORITHMS.getId()
+                    );
+                }
+            }
+        }
+    }
+
+    /**
+     * Determines if a variable reference refers to an IssuerConfig record with signatureConfig.algorithm == NONE.
+     * It walks up the AST to check function bodies and module-level declarations.
+     *
+     * @param varRef the simple name reference node pointing to the variable
+     * @return true if the variable's initializer mapping contains algorithm NONE, false otherwise
+     */
+    private boolean isInsecureVariableReference(SimpleNameReferenceNode varRef) {
+        String varName = varRef.name().text();
+        Node current = varRef.parent();
+        while (current != null) {
+            // Case 1: variable declared inside a function body
+            if (current instanceof FunctionBodyBlockNode body) {
+                for (StatementNode stmt : body.statements()) {
+                    if (stmt instanceof VariableDeclarationNode varDecl
+                            && varDecl.typedBindingPattern().bindingPattern() instanceof CaptureBindingPatternNode cap
+                            && cap.variableName().text().equals(varName)
+                            && varDecl.initializer().isPresent()
+                            && varDecl.initializer().get() instanceof MappingConstructorExpressionNode mappingCtor) {
+                        if (isInsecureInlineMappingLiteral(mappingCtor)) {
+                            return true;
+                        }
+                    }
+                }
+            }
+            // Case 3: variable declared at module level
+            if (current instanceof ModulePartNode module) {
+                for (ModuleMemberDeclarationNode member : module.members()) {
+                    if (member instanceof ModuleVariableDeclarationNode varDecl
+                            && varDecl.typedBindingPattern().bindingPattern() instanceof CaptureBindingPatternNode cap
+                            && cap.variableName().text().equals(varName)
+                            && varDecl.initializer().isPresent()
+                            && varDecl.initializer().get() instanceof MappingConstructorExpressionNode mappingCtor
+                            && isInsecureInlineMappingLiteral(mappingCtor)) {
+                        return true;
+                    }
+                }
+            }
+            current = current.parent();
```

Comment:
I think we can get directly the invokable node of a given node from the semantic model, Please check